### ISSUE TITLE
Upgrade 1.3.2 to 1.4.2 [BUG]

### ISSUE THREAD

**Issue by BinhPhuongIT** [longhorn/longhorn#6122]
hi team, after i upgrade longhorn 1.3.2 to 1.4.2 according to instructions, every action from Longhorn UI gives 403 error
![image](https://github.com/longhorn/longhorn/assets/69758987/0d5f1f13-ff4e-4adf-8707-2e49df10d25b)

**Comment by innobead** [comment]
Are longhorn components running well?

**Comment by BinhPhuongIT** [comment]
still running fine. only 403 error when manipulating from the UI

**Comment by BinhPhuongIT** [comment]
I've been able to handle it.
cause has ingress configuration.

you ask me more. After upgrading from 1.3.2 to 1.4.2, longhorn-engine:v1.3.2 is still retained. When I delete longhorn-engine:v1.3.2, I get an error

![image](https://github.com/longhorn/longhorn/assets/69758987/e27f47f9-af19-4087-9a29-deae6566f047)

**Comment by innobead** [comment]
> I've been able to handle it. cause has ingress configuration.

What ingress configuration caused that? 

> you ask me more. After upgrading from 1.3.2 to 1.4.2, longhorn-engine:v1.3.2 is still retained. When I delete longhorn-engine:v1.3.2, I get an error
> 
> ![image](https://user-images.githubusercontent.com/69758987/245674074-e27f47f9-af19-4087-9a29-deae6566f047.png)

What error did you get? did you notice any error info from longhorn manager log?

cc @mantissahz to follow up

**Comment by mantissahz** [comment]
Hi @BinhPhuongIT,
Could you describe your cluster more details? k8s version, os, how did you install Longhorn and how did you upgrade Longhorn?
Yeah, as @innobead mentioned, any error info from `longhorn-manager` or `longhorn-ui` logs?

**Comment by mantissahz** [comment]
Yes, if you did not set the setting `Concurrent Automatic Engine Upgrade Per Node Limit
` to [automatically upgrade engines](https://longhorn.io/docs/1.4.2/deploy/upgrade/auto-upgrade-engine/), you need to [upgrade engine manually](https://longhorn.io/docs/1.4.2/deploy/upgrade/upgrade-engine/)
You could see the reference count for the engine that volume is using.

And Does your Longhorn UI work fine? 

>I've been able to handle it. cause has ingress configuration.

What ingress configuration caused that? just curious

**Comment by BinhPhuongIT** [comment]
![image](https://github.com/longhorn/longhorn/assets/69758987/b7306ba6-0495-4005-bf47-b1033ab4a302)

**Comment by mantissahz** [comment]
There are something synchronous operations that will use the longhorn-engine image. It would get some errors if you deleted it unexpectedly.
The better way is that deleting the old longhorn-engine image after upgrading all volume's longhorn-engine to new version.
And I think Longhorn should have some protection if there are volume using old longhorn-engine image.
